### PR TITLE
Implement home screen letter dictionary

### PR DESCRIPTION
**Pull request by samymessouak**
### Summary

This pull request introduces several key changes to improve the usability and reduce code duplication in the HomeScreen:

### Changes Implemented

1. **Add Scrolling Feature for Letter Dictionary:**
   - Implemented a scrollable Letter Dictionary.
   - Added functionality to cycle through letters from 'a' to 'z' using left and right arrow buttons.
   - Wrapped the letter and associated gesture in state to enable the navigation.
   - Created `getLetterIconResId()` utility function to map each letter to its respective drawable resource.

2. **Unit Tests for Scrolling Feature:**
   - Added unit tests to verify that the correct letter and gesture icon are displayed after each arrow click.
   - Added test tags to the Letter Dictionary composables for better testability.

3. **Refactored Util Buttons:**
   - Replaced the following buttons with the `UtilButton` reusable composable to avoid duplicated code:
     - Quest Button
     - Camera Toggle Button
     - Info Button
   - Reduced the duplicated implementation by using the `Utils` class.

### Next Steps
- Ensure the scrollable dictionary's behavior remains intuitive for users of all ages.
- Review test coverage to make sure that all edge cases are addressed.

image

### CODE REVIEW

**Comment by ThomasMPont** [comment]
I felt free to commit a quick refactor. I felt the function you created to get letter icons belonged in Utils.kt.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt
Comment (ThomasMPont) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt:1]: Good job improving the coverage on the home screen. As you said, some further more complex tests could be written to test the new code's robustness.

File: app/src/main/java/com/github/se/signify/ui/Utils.kt
Comment (ThomasMPont) [inline app/src/main/java/com/github/se/signify/ui/Utils.kt:1]: This feels unnecessary, but I could not find a better way to do it. It'll have to do for now.

File: app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt
Comment (ThomasMPont) [inline app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt:1]: I see you have replaced some of the buttons from this screen with new common elements. As the coaches said, we should avoid making changes that are not related to a branch's issue. Since this should not raise merge conflicts right now, I'll let it slide.

**Review by ThomasMPont — APPROVED** [review]
Good job on the letter dictionary! It seems functional and was made in accordance with its planned look on Figma. It also raises the overall coverage of the screen by quite a bit.

I see you have gone and replaced some of the screen's buttons with new common elements from Utils.kt. To stick to the coaches instructions, you should either have:
- Edited the task to also include updating the screen's other elements.
(e.g. Update home screen...)
- Done the changes in a separate task.

We all know the feeling of wanting to fix blatant problems we see asap, but taking the time to do the changes separately is crucial.

Since nobody else seems to be working on the home screen at the moment, I think it should be fine to merge this anyway, exceptionally.

File: app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt [hunk 1]
```diff
@@ -107,4 +108,56 @@ class HomeScreenTest {
       verify(onClick).invoke(exercise) // Verify onClick was called with the exercise
     }
   }
+
+  @Test
```
Comment (ziyadbagh) [inline app/src/androidTest/java/com/github/se/signify/ui/screens/HomeScreenTest.kt:112]: Good job.

**Review by ziyadbagh — APPROVED** [review]
